src/lib/resources/mail/flags.ts

```diff
@@ -26,3 +24,27 @@ export const {
   },
   shortIDName: "mail address",
 });
+
+export const {
+  flags: mailDeliveryBoxFlags,
+  args: mailDeliveryBoxArgs,
+  withId: withDeliveryBoxId,
+} = makeProjectFlagSet("maildeliverybox", "d", {
+  normalize: async (apiClient, projectId, id): Promise<string> => {
+    if (validateUuid(id)) {
+      return id;
+    }
+
+    const response = await apiClient.mail.listDeliveryBoxes({ projectId });
+    assertStatus(response, 200);
+
+    const deliveryBox = response.data.find(
+      (deliveryBox) => deliveryBox.id === id,
+    );
+    if (!deliveryBox) {
+      throw new Error(`mail delivery box with id "${id}" not found`);
+    }
+
+    return deliveryBox.id;
+  },
+});
```

Comment:
Yes, using `makeProjectFlagSet` would make sense for a short-id lookup -- and that only in cases where it's not supported natively by the API. Currently it's used for domain/DNS names (when a project is set in the context, you can use a domain name instead of the domain ID).